airbyte-ci/connectors/pipelines/pipelines/builds/python_connectors.py

```diff
@@ -15,13 +19,102 @@ class BuildConnectorImage(BuildConnectorImageBase):
+
+        if connector_base_image not in ALL_BASE_IMAGES:
+            return StepResult(
+                self,
+                StepStatus.FAILURE,
+                f"Connector base image {connector_base_image} does not exists. " f"Supported connector base images are {ALL_BASE_IMAGES}",
+            )
+        base: Container = ALL_BASE_IMAGES[connector_base_image](self.dagger_client).container
+        connector: Container = await self._build_from_base(base)
         try:
             return await self.get_step_result(connector.with_exec(["spec"]))
         except QueryError as e:
             return StepResult(self, StepStatus.FAILURE, stderr=str(e))
 
+    async def _build_from_base(self, base: Container):
+        pip_cache: CacheVolume = self.context.dagger_client.cache_volume("pip_cache")
+        snake_case_name = self.context.connector.technical_name.replace("-", "_")
+
+        setup_dependencies_to_mount = await find_local_python_dependencies(
+            self.context,
+            str(self.context.connector.code_directory),
+            search_dependencies_in_setup_py=True,
+            search_dependencies_in_requirements_txt=False,
+        )
+
+        with_setup_file: Container = (
+            base.with_env_variable("DAGGER_BUILD", "True")
+            .with_workdir("/airbyte/integration_code")
+            .with_mounted_cache("/root/.cache/pip", pip_cache)
+            .with_file("setup.py", (await self.context.get_connector_dir(include="setup.py")).file("setup.py"))
+        )
+        with_local_dependencies = with_setup_file
+
+        for dependency_path in setup_dependencies_to_mount:
+            in_container_dependency_path = f"/local_dependencies/{Path(dependency_path).name}"
+            with_local_dependencies = with_local_dependencies.with_mounted_directory(
+                in_container_dependency_path, self.context.get_repo_dir(dependency_path)
+            )
+        with_local_dependencies_installed = with_local_dependencies.with_exec(["pip", "install", "--prefix=/usr/local", "."])
+        with_main = with_local_dependencies_installed.with_file("main.py", (await self.context.get_connector_dir()).file("main.py"))
+        with_connector_code = with_main.with_directory(
+            snake_case_name, (await self.context.get_connector_dir(include=snake_case_name)).directory(snake_case_name)
+        )
+        connector_container = (
+            with_connector_code.with_env_variable("AIRBYTE_ENTRYPOINT", " ".join(self.entrypoint))
+            .with_entrypoint(self.entrypoint)
+            .with_label("io.airbyte.version", self.context.metadata["dockerImageTag"])
+            .with_label("io.airbyte.name", self.context.metadata["dockerRepository"])
+        )
+        return await self.finalize_build(self.context, connector_container)
+
+    @staticmethod
+    async def finalize_build(context: ConnectorContext, connector_container: Container) -> Container:
```

Comment:
This code is currently used in the `environment` module: https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/actions/environments.py#L893